### PR TITLE
Implement model types for official pytorch models

### PR DESCRIPTION
**Pull request by eddyxu**
Add all official pytorch object detections models as well as classification models like ResNet and EfficientNet.

### CODE REVIEW

**Comment by eddyxu** [comment]
Closes https://github.com/eto-ai/rikai/issues/551

File: python/rikai/pytorch/models/__init__.py [hunk 1]
```diff
@@ -13,41 +13,15 @@
 #  limitations under the License.
 
 """Rikai-implemented PyTorch models and executors."""
-from abc import ABC
-from typing import Optional
 
-import torch
-
```
Comment (eddyxu) [inline python/rikai/pytorch/models/__init__.py:20]: Moved out to address cyclic dependency

File: python/tests/pytorch/models/test_model_types.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    for name in [
+        "fasterrcnn",
+        "fasterrcnn_resnet50_fpn",
+        "fasterrcnn_mobilenet_v3_large_fpn",
+        "fasterrcnn_mobilenet_v3_large_320_fpn",
+    ]:
+        spark.sql(
+            f"""CREATE OR REPLACE MODEL {name}
+            FLAVOR pytorch
+            MODEL_TYPE fasterrcnn_mobilenet_v3_large_fpn
```
Comment (changhiskhan) [inline python/tests/pytorch/models/test_model_types.py:35]: is this actually testing all 4 model types?

File: python/rikai/pytorch/models/efficientnet.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+"""
+EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks
+https://arxiv.org/abs/1905.11946
+"""
+
+import torchvision
+
+from rikai.pytorch.models.torch import ClassificationModelType
+
+efficientnet_b0 = ClassificationModelType(
```
Comment (changhiskhan) [inline python/rikai/pytorch/models/efficientnet.py:24]: are these model types tested?

File: python/rikai/pytorch/models/keypointrcnn.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import torchvision
+
+from rikai.pytorch.models.torch import ObjectDetectionModelType
+
+keypointrcnn = ObjectDetectionModelType(
```
Comment (changhiskhan) [inline python/rikai/pytorch/models/keypointrcnn.py:19]: this and also maskrcnn are untested?

Comment (eddyxu) [reply python/rikai/pytorch/models/keypointrcnn.py:19]: added tests for these models (keypointrcnn, masksrcnn, resnet , efficientnet)

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**